lib/_http_agent.js

```diff
@@ -120,12 +129,15 @@ function Agent(options) {
           socket[async_id_symbol] = -1;
           socket._httpMessage = null;
           this.removeSocket(socket, options);
+          socket.once('error', freeSocketErrorListener);
           freeSockets.push(socket);
         } else {
           // Implementation doesn't want to keep socket alive
+          socket.once('error', nop);
           socket.destroy();
         }
       } else {
+        socket.once('error', nop);
```

Comment:
These `socket.once('error', nop);` are not directly related to this PR and could be moved into separate PR.

Reply:
@lpinca Might not like this but I can't think of any meaningful way not to swallow errors here. Any input?

Reply:
I think I understand why you add those no-op error listeners but I find it really hard to reason about whether it's correct to add them or not. Why are they necessary?

Reply:
at this point there is no error handler, but they socket might still fail after the call to `destroy` causing an unhandled exception

Reply:
Really don't like the non-ops that swallow errors. What is the behavior you're looking to achieve here?

Reply:
I think those are overall ok because of what @ronag said in https://github.com/nodejs/node/pull/32003/files#r385724918. They are not really relevant to this PR either, so they can be moved to another one if you prefer @jasnell.

Reply:
@jasnell: I don't think I can explain it better than my previous comment. No one owns the socket here and it can emit `'error'`.

Reply:
Since no one is writing to this socket how can it emit `'error'`? on destroy?

Reply:
It can emit an error when destroying the handle? I know of no guarantee that it won't emit any errors here.

Reply:
Also I believe ECONNRESET and the alike is possible whenever?

Reply:
I don't think it can emit errors while`_handle.close()` does its job.

Reply:
No, those occur when you write to a socket that has been destroyed by the other peer.

Reply:
Still, this assumption relies quite heavily on an implementation detail. Nothing in the socket api says that destroy cannot fail...

I did not expect this to be a controversial change. Should I remove it from this PR and open another one where we can continue this discussion?

Reply:
Removed this part for now.